### PR TITLE
Add power management tutorial to power_manage.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Add power management tutorial with content provided by the Evangelism team.

### CODE REVIEW

**Comment by AnotherButler** [comment]
@bulislaw @evedon This is backlog improvement content we received from the Evangelism team. If someone from your team could please review this for accuracy at their convenience, I would appreciate it.

File: docs/tutorials/optimize/power/power_manage.md [hunk 1]
```diff
@@ -1,3 +1,271 @@
...
+
+## Power modes
+
+Mbed OS contains three power modes:
+
+- Active - The MCU and all clocks are running.
+- Sleep - The core system clock is disabled. This eliminates dynamic power that the processor, memory systems and buses use.
+- Deep sleep - In addition to the core system clock, all high-frequency clocks are disabled, and the [SysTick](../apis/rtos.html) is disabled.
+
+Switching between these power modes occurs automatically. When all threads in the system are idle, Mbed OS yields control to the [idle thread](../apis/idle-loop.html). The idle thread then invokes the sleep manager, which brings the system to sleep or deep sleep mode. The idle thread also sets a timer to wake up the system again, but you can also wake up the system through an external interrupt or the Real-Time Clock (RTC).
```
Comment (bulislaw) [inline docs/tutorials/optimize/power/power_manage.md]: Mbed OS doesn't support waking of the RTC, can we replace it with low power ticker.

File: docs/tutorials/optimize/power/power_manage.md [hunk 2]
```diff
@@ -1,3 +1,271 @@
...
+   - High-speed clocks should be turned off.
+   - RTC is running and keeps time.
+
+Both HAL sleep functions work like an Arm Wait For Interrupt (WFI) instruction, where the function returns when there is a pending interrupt. To achieve this, the sleep manager calls these functions from a [critical section](../apis/criticalsectionlock.html). Often (though this is device specific) `hal_sleep` is implemented as just a `__WFI()` call, and deep sleep is the same call but surrounded by power control settings that limit the wake-up sources and functioning peripherals.
+
+This is also why the MCU wakes up from sleep every millisecond when tickless is not enabled. In nontickless mode, SysTick needs to fire every millisecond and does this by setting an interrupt on the usticker. Right after the SysTick, the sleep manager puts the MCU back to sleep. However, this also means that in nontickless mode, you can't put the MCU in deep sleep because the wake-up latency is bigger than the SysTick interval.
+
+For more information on the design of tickless and the sleep manager, please see the [office hours video with Bartek Szwatkowski](https://www.youtube.com/watch?v=OFfOlBaegdg).
```
Comment (bulislaw) [inline docs/tutorials/optimize/power/power_manage.md]: Szatkowski ;)

File: docs/tutorials/optimize/power/power_manage.md [hunk 3]
```diff
@@ -1,3 +1,271 @@
...
+
+This is related to the maximum timeout of the hardware low power ticker. It can only be asleep accurately for one second, so the device wakes up every second, and Mbed OS brings the device back into deep sleep afterward. You often can override this behavior by choosing a different clock source, but this is device specific. Look at the data sheet for your MCU. On the DISCO-L475VG-IOT01A1, you can override this by adding the following line to the `target_overrides` section of your `mbed_app.json`:
+
+```json
+        "target.lpticker_lptim_clock": 4
+```
+
+### Device does not sleep in bare-metal mode
+
+The sleep manager does not load when running Mbed OS in bare-metal mode. We may add this capability in a future release.
```
Comment (bulislaw) [inline docs/tutorials/optimize/power/power_manage.md]: we should add something like "you should manually call the `sleep()` function making sure you set up the wake up source`

**Review by bulislaw — COMMENTED** [review]
That looks good. Couple of nits + we should link the power management api page from this tutorial and the other way around as well.

**Comment by AnotherButler** [comment]
This is failing Travis. Could someone please check the code snippets?

**Comment by evedon** [comment]
@gpsimenos Please take a loot at code snippets

**Comment by AnotherButler** [comment]
ping @gpsimenos

**Review by gpsimenos — APPROVED** [review]
Code snippets look good

**Comment by evedon** [comment]
@gpsimenos look why travis code-snippets is failing

**Comment by gpsimenos** [comment]
> @gpsimenos look why travis code-snippets is failing

There is no reason for it to fail. Where is this script located? Maybe I can debug it.

**Comment by AnotherButler** [comment]
@gpsimenos Maybe they're not supposed to compile, in which case we should add "NOCI" to them? The script is here: https://github.com/ARMmbed/mbed-os-5-docs/blob/development/check_tools/find_bad_code_snippets.sh
Thanks for looking into this 👍

**Comment by AnotherButler** [comment]
ping @gpsimenos

**Comment by gpsimenos** [comment]
I had a look at the script, still don't see why it would fail. Let's add NOCI to that snippet.

**Pull request merged**